### ISSUE TITLE
3D pose training data for demo

### ISSUE THREAD

**Issue by momo1986** [tensorflow/graphics#12]
Hello, Google Guys.
I am very interested in your 3D pose reconstruction project.
In your demo,
https://colab.research.google.com/github/tensorflow/graphics/blob/master/tensorflow_graphics/notebooks/6dof_alignment.ipynb
It looks like that the data is generated randomly.

For my own transfer, I am very interested that whether there is a corresponding training dataset?

Thus, I can fabricate or produce my own dataset for my specific task.

Thanks  & Regards!

**Comment by julienvalentin** [comment]
Hi!

In the demo you refer to, we sample the 6 degrees of freedom (rotation and translation) that we wish the network to be able to recover. The geometry of the object (mesh) is fixed.
I am not exactly sure to understand what you are after, could you detail a bit more?

Best.

**Comment by julienvalentin** [comment]
Let us know @momo1986 ;)

**Comment by AndriiTsok** [comment]
Hello @julienvalentin, we are currently reviewing TF-Graphics as a solution for a new NN architecture.
We have a dataset of 2d pictures (projections of one type of an object) of 3d object with corresponding R|T matrix. Would you recommend what parts of the TFG we could incorporate for building such a regressor network? Is TFG would be a good choice for such task?

**Comment by julienvalentin** [comment]
@AndriiTsok I would encourage you to have a look at our demo on pose estimation:
https://colab.sandbox.google.com/github/tensorflow/graphics/blob/master/tensorflow_graphics/notebooks/6dof_alignment.ipynb

Also, could you say a bit more about the exact setting and how you want to approach the problem + rotation formalism you plan on using?
One solution that I would encourage to try is a cascaded approach, where each level of the cascade predicts an offset to the solution (R|t) obtained by the previous level. Usually 2-3 levels are good enough for quite a few problems.

**Comment by AndriiTsok** [comment]
@julienvalentin thank you so much for your reply.

At the moment we have a dataset of pictures of simple shapes and labels - (R|t) matrices for 3d objects that were used to create these pictures (rendering). What we are planning todo is to create a regressor that predicts these (R|t) matrices. We have a few loss functions in mind L2 loss on the
quaternion and translation, IoU, others. The idea is that the network will be able to predict pose - (R|t) matrix to fit 3d model to the same position as it is on a picture. We plan to run this network architecture on edge devices.

@julienvalentin After your note about cascaded approach I have got quite a few ideas for our solution. But could you please describe a little bit more about this approach, so we could research more on this one. As it is currently not clear how it would work. Thanks!

**Comment by julienvalentin** [comment]
It seems that this can safely be close. Feel free to re-open if needed.
